### PR TITLE
Scorpius-19.6.0-PLAT-23876 support search_term aggregation and export…

### PR DESCRIPTION
**Pull request by salma2222**
…tocsv in search history plugin

### CODE REVIEW

File: deployment/updates/scripts/add_permissions/2023_05_07_add_searchhistory_permissions.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+/**
+ * @package deployment
+ * @subpackage naos.roles_and_permissions
```
Comment (inbal-ben-david) [inline deployment/updates/scripts/add_permissions/2023_05_07_add_searchhistory_permissions.php:4]: Scorpius

Comment (salma2222) [reply deployment/updates/scripts/add_permissions/2023_05_07_add_searchhistory_permissions.php]: done

File: plugins/search_history/lib/model/filters/ESearchHistoryFilter.php [hunk 2]
```diff
@@ -10,6 +10,12 @@ class ESearchHistoryFilter extends ESearchBaseFilter
 	const STARTS_WITH_PAGE_SIZE = 100;
 	const DEFAULT_LIST_SIZE = 500;
 	const MAX_SEARCH_TERM_LENGTH = 64;
+	const MAX_AGGREGATION_SIZE = 100;
+	const LIMIT_IN_MONTHS = 6;
+	const SECONDS_IN_DAY = 24*60*60;
+	const DAYS_IN_MONTH = 31;
```
[the remark this answers is not in the document]
Comment (salma2222) [reply plugins/search_history/lib/model/filters/ESearchHistoryFilter.php]: changed the name

File: plugins/search_history/lib/model/filters/ESearchHistoryFilter.php [hunk 3]
```diff
@@ -86,19 +180,94 @@ protected function applyFilter()
...
-		$this->query[kESearchQueryManager::SORT_KEY] = array(ESearchHistoryFieldName::TIMESTAMP => array(kESearchQueryManager::ORDER_KEY => ESearchSortOrder::ORDER_BY_DESC));
-		$this->query[kESearchQueryManager::FROM_KEY] = 0;
-		$this->query[kESearchQueryManager::SIZE_KEY] = $pageSize;
+
+	}
+
+	protected function validateTimestampLimit()
+	{
+		$limitInMonths = $this->getAggregationTimestampLimit();
+		$ceilingTimestamp = $this->getCeilingTimestamp();
```
[the remark this answers is not in the document]
Comment (salma2222) [reply plugins/search_history/lib/model/filters/ESearchHistoryFilter.php]: changed

File: plugins/search_history/lib/model/filters/ESearchHistoryFilter.php [hunk 4]
```diff
@@ -64,19 +112,65 @@ public function execQueryFromFilter()
...
+		}
+
+		$this->applyFilterFields($boolQuery, true);
+
+		$aggregationKey = ESearchHistoryAggregationItem::KEY . ':' . $this->aggregation->getFieldName();
+		$this->aggregation->setSize($this->getAggregationSize());
+		$aggregations[$aggregationKey] = $this->aggregation->getAggregationCommand();
+
+		$this->query[ESearchAggregations::AGGS] = $aggregations;
+		$this->query[kESearchQueryManager::SIZE_KEY] = 0;
```
[the remark this answers is not in the document]
Comment (salma2222) [reply plugins/search_history/lib/model/filters/ESearchHistoryFilter.php:162]: added

File: plugins/search_history/batch/KExportSearchHistoryEngine.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+{
+
+	public function fillCsv(&$csvFile, &$data)
+	{
+		KalturaLog::info('Exporting search history content');
+		$filter = clone $data->filter;
+
+		try
+		{
+			$searchHistoryList = KBatchBase::$kClient->searchHistory->listAction($filter);
```
[the remark this answers is not in the document]
Comment (salma2222) [reply plugins/search_history/batch/KExportSearchHistoryEngine.php:17]: added a comment

File: plugins/search_history/SearchHistoryPlugin.php [hunk 6]
```diff
@@ -2,46 +2,143 @@
...
+			}
+		}
+
+		return null;
+
+	}
+
+	public static function getEnums($baseEnumName = null)
+	{
+		if (is_null($baseEnumName))
```
[the remark this answers is not in the document]
Comment (salma2222) [reply plugins/search_history/SearchHistoryPlugin.php:131]: done

File: plugins/search_history/SearchHistoryPlugin.php [hunk 7]
```diff
@@ -2,46 +2,143 @@
...
+		if ($baseClass == 'KalturaJobData')
+		{
+			if ($enumValue == self::getApiValue(SearchHistorySearchTermCsvBatchType::SEARCH_TERMS_CSV) ||
+				$enumValue == self::getBatchJobTypeCoreValue(SearchHistorySearchTermCsvBatchType::SEARCH_TERMS_CSV))
+			{
+				return new KalturaSearchHistoryCsvJobData();
+			}
+		}
+
+		if($baseClass == 'KalturaJobData' && $enumValue == BatchJobType::EXPORT_CSV && (isset($constructorArgs['coreJobSubType']) &&  $constructorArgs['coreJobSubType']== self::getExportTypeCoreValue(SearchHistoryExportObjectType::SEARCH_TERM)))
```
[the remark this answers is not in the document]
Comment (salma2222) [reply plugins/search_history/SearchHistoryPlugin.php]: done

**Pull request merged**